symmetric/src/serializing_hasher.rs

```diff
@@ -1,16 +1,28 @@
+use core::borrow::Borrow;
+use core::marker::PhantomData;
+
 use p3_field::{PrimeField32, PrimeField64};
+use serde::{Deserialize, Serialize};
 
 use crate::CryptographicHasher;
 
-/// Maps input field elements to their 4-byte little-endian encodings, and maps output of the form
-/// `[u8; 32]` to `[F; 8]`.
+/// Maps input field elements to their 4-byte little-endian encodings, outputs `[u8; 32]`.
 #[derive(Copy, Clone)]
 pub struct SerializingHasher32<Inner> {
     inner: Inner,
 }
 
-/// Maps input field elements to their 8-byte little-endian encodings, and maps output of the form
-/// `[u8; 32]` to `[F; 4]`.
+/// A wrapper around an array digest, with a phantom type parameter to ensure that the digest is
+/// associated with a particular field.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(bound(serialize = "[W; DIGEST_ELEMS]: Serialize"))]
+#[serde(bound(deserialize = "[W; DIGEST_ELEMS]: Deserialize<'de>"))]
+pub struct Hash<F, W, const DIGEST_ELEMS: usize> {
```

Comment:
Seems this gets used whether or not SerializingHasher is used, so maybe it should get its own file/module